autokeras/adapters/input_adapter.py

```diff
@@ -99,30 +57,18 @@ def _check(self, x):
             raise TypeError('Expect the data to TextInput to be strings, but got '
                             '{type}.'.format(type=x.dtype))
 
-    def _convert_to_dataset(self, x):
+    def convert_to_dataset(self, x):
+        if len(x.shape) == 1:
+            x = x.reshape(-1, 1)
         if isinstance(x, np.ndarray):
             x = tf.data.Dataset.from_tensor_slices(x)
         return x
 
 
-class StructuredDataInput(Input):
-    """Input node for structured data.
-
-    The input data should be numpy.ndarray, pandas.DataFrame or tensorflow.Dataset.
-    The data should be two-dimensional with numerical or categorical values.
+class StructuredDataInputAdapter(adapter_module.Adapter):
 
-    # Arguments
-        column_names: A list of strings specifying the names of the columns. The
-            length of the list should be equal to the number of columns of the data.
-            Defaults to None. If None, it will be obtained from the header of the csv
-            file or the pandas.DataFrame.
-        column_types: Dict. The keys are the column names. The values should either
-            be 'numerical' or 'categorical', indicating the type of that column.
-            Defaults to None. If not None, the column_names need to be specified.
-            If None, it will be inferred from the data. A column will be judged as
-            categorical if the number of different values is less than 5% of the
-            number of instances.
-    """
+    CATEGORICAL = 'categorical'
+    NUMERICAL = 'numerical'
```

Comment:
I would recommend not having global keys like this be class attributes. Instead, make it external constants.